### DISCUSSION TITLE
CentoSync: A Web-Based Video Player with MTC/SPP Sync Powered by WebMidi.js

### DISCUSSION THREAD

**Discussion by centomila** [djipco/webmidi#443] [Show and tell]
# CentoSync: A Web-Based Video Player with MTC/SPP Sync Powered by WebMidi.js

Hello WebMidi.js community! I'm excited to share a project that wouldn't have been possible without this fantastic library. Introducing CentoSync, a web application that synchronizes video playback with your DAWs using MIDI Time Code (MTC) or MIDI Clock with Song Position Pointer (SPP).

## Project Overview

CentoSync leverages the power of WebMidi.js to enable seamless communication between web browsers and DAWs. It's designed to solve the common problem of video synchronization in audio production workflows.

### Key Features:
- Real-time synchronization using MTC or MIDI Clock with SPP
- Compatible with multiple DAWs (tested with Bitwig, Ableton Live, FL Studio, and more)
- Video offset adjustment
- Detailed video information display

### Technical Highlights:
- Utilizes WebMidi.js for MIDI input processing
- Implements both MTC and SPP synchronization algorithms
- Handles various MIDI message types (Quarter Frame, Full Frame, Clock, Start, Stop, Continue)
- Local storage for user settings persistence

## WebMidi.js Implementation

WebMidi.js plays a crucial role in CentoSync's functionality. Here are some key areas where it's utilized:

1. MIDI input detection and selection
2. Real-time MIDI message parsing
3. Handling of system exclusive messages for full MTC frames
4. Processing of MIDI Clock messages for SPP synchronization

## Try It Out

👉👉👉 **CentoSync is freely available at: https://centosync.centomila.com/** 👈👈👈

I welcome feedback from the WebMidi.js community, especially regarding MIDI handling and performance optimizations.

## More Information
- Project Website: https://centomila.com/software/centosync-video-player-with-mtc-sync/
- Source Code: https://github.com/centomila/CentoSync-VideoPlayer-With-MTC-Sync

## Contribution and Feedback

As an open-source project, CentoSync is open to contributions. If you're interested in improving MIDI handling, adding features, or optimizing performance, please feel free to check out the GitHub repository.

I'm particularly interested in feedback on:
1. MIDI timing accuracy
2. Cross-browser compatibility
3. Performance optimizations for MIDI message processing

Thank you to the WebMidi.js team and community for creating and maintaining such a powerful tool for web-based MIDI applications!

👋

**Comment by djipco** [comment]
Wow, this looks awesome! Thanks for sharing. I will definitely look it up.
